src/QMCDrivers/WFOpt/QMCCostFunction.cpp

```diff
@@ -537,8 +537,9 @@ QMCCostFunction::Return_rt QMCCostFunction::correlatedSampling(bool needGrad)
   for (int ip = 0; ip < NumThreads; ++ip)
   {
     //    synchronize the random number generator with the node
-    (*MoverRng[ip]) = (*RngSaved[ip]);
-    hClones[ip]->setRandomGenerator(MoverRng[ip]);
+    if (MoverRng[ip] && RngSaved[ip])
```

Comment:
Could you replace this if-test with an assert.  If the dereference is protected here, the state of the saved RNG won't get copied to the working RNG.   This  either puts off the segfault (if MoverRng is NULL), or creates a very subtle bug (if RngSaved is NULL)
This is the important line that resets the state of the RNG for the pseudopotential grid offset, so that the same points are used here as in checkConfigurations.